### PR TITLE
mtx3: add getOperatorParametersExtended, setOperatorParametersExtended commands

### PR DESCRIPTION
**Pull request by aleks-and**

### CODE REVIEW

File: src/mtx3/utils/CommandBinaryBuffer.ts [hunk 1]
```diff
@@ -1101,7 +1101,71 @@ export interface IOperatorParameters {
...
+     */
+    RESET_DAY_MAX_POWER_KEY: boolean,
+
+    /**
+     * Reset monthly maximum power using button.
+     */
+    RESET_MONTH_MAX_POWER_KEY: boolean,
+
+    /**
+     * `1` - optoport is unlocked by button, `0` - optoport is unlocked.
```
Comment (DeadByElpy) [inline src/mtx3/utils/CommandBinaryBuffer.ts:1116]: boolean can't handle numeric type

File: src/mtx3/utils/CommandBinaryBuffer.ts [hunk 2]
```diff
@@ -1101,7 +1101,71 @@ export interface IOperatorParameters {
     displaySet4: IDisplaySet4OperatorParameter
 }
 
+export interface IDefine1OperatorParameterExtended {
+    /**
+     * Reset daily maximum power using button.
```
Comment (DeadByElpy) [inline src/mtx3/utils/CommandBinaryBuffer.ts:1106]:
```suggestion
     * Reset daily maximum power.
```

or add additional description about what button and how it relates to parameter in codecs

File: src/mtx3/utils/CommandBinaryBuffer.ts [hunk 3]
```diff
@@ -1101,7 +1101,71 @@ export interface IOperatorParameters {
...
+     * Allow display indication in battery mode (`1` - enabled).
+     *
+     * since build `0.0.17`
+     */
+    ALLOW_BROWNOUT_INDICATION: boolean
+}
+
+export interface IOperatorParametersExtended {
+    /**
+     * Timeout for automatic relay activation based on `IMAX`, `PMAX`, `IDIFF`, `COSFI`, minutes.
```
Comment (DeadByElpy) [inline src/mtx3/utils/CommandBinaryBuffer.ts:1137]: need to discuss

File: src/mtx3/utils/CommandBinaryBuffer.ts [hunk 4]
```diff
@@ -1101,7 +1101,71 @@ export interface IOperatorParameters {
...
+     */
+    define1: IDefine1OperatorParameterExtended,
+
+    /**
+     * Timeout for relay activation based on `IMAX`, `PMAX`, `IDIFF`, `COSFI`, seconds.
+     */
+    timeoutRelayKey: types.TUint8,
+
+    /**
+     * Timeout for relay activation upon restoration of quality voltage, seconds.
```
Comment (DeadByElpy) [inline src/mtx3/utils/CommandBinaryBuffer.ts:1152]:
```suggestion
     * Timeout for relay activation upon restoration of quality voltage, seconds.
```

voltage or seconds, or "in seconds"?

**Review by DeadByElpy — CHANGES_REQUESTED** [review]

File: src/mtx3/commands/downlink/setOperatorParametersExtended.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ *     define1: {
+ *         RESET_DAY_MAX_POWER_KEY: false,
+ *         RESET_MONTH_MAX_POWER_KEY: false,
+ *         BLOCK_KEY_OPTOPORT: false,
+ *         MAGNET_SCREEN_CONST: false,
+ *         ALLOW_BROWNOUT_INDICATION: false
+ *     },
+ *     timeoutRelayKey: 0,
+ *     timeoutRelayAuto: 5,
+ *     reserved1: 0,
```
[the remark this answers is not in the document]
Comment (aleks-and) [reply src/mtx3/commands/downlink/setOperatorParametersExtended.ts]: fixed

File: src/mtx3/utils/CommandBinaryBuffer.ts (hunk 1 above)
Comment (aleks-and) [reply src/mtx3/utils/CommandBinaryBuffer.ts]: fixed

File: src/mtx3/utils/CommandBinaryBuffer.ts (hunk 2 above)
Comment (aleks-and) [reply src/mtx3/utils/CommandBinaryBuffer.ts]: fixed

File: src/mtx3/utils/CommandBinaryBuffer.ts (hunk 3 above)
Comment (aleks-and) [reply src/mtx3/utils/CommandBinaryBuffer.ts]: fixed

**Review by DeadByElpy — APPROVED** [review]

**Review by Pnom — APPROVED** [review]

**Pull request merged**